base/shared/src/main/scala/scalaz/data/IList.scala

```diff
@@ -2,12 +2,21 @@ package scalaz
 package data
 
 import typeclass.IsCovariantClass
+import scala.annotation.tailrec
 
 trait IListModule {
   type IList[A]
 
   def uncons[A](as: IList[A]): Maybe2[A, IList[A]]
 
+  def cons[A](a: A, list: IList[A]): IList[A]
+
+  def concat[A](as1: IList[A], as2: IList[A]): IList[A]
+
+  def empty[A]: IList[A]
```

Comment:
`empty` and `concat` are very valuable identifiers. Should they be ceded to `IList`, or should we make users utilize `Monoid` functions for `IList`?

Reply:
I have an idea here for later: we stop providing the `*Functions` traits in the prelude, and instead we put a bunch of functions inside a new `Z` object.